src/hotspot/share/runtime/javaThread.hpp

```diff
@@ -1145,6 +1146,16 @@ class JavaThread: public Thread {
   void interrupt();
   bool is_interrupted(bool clear_interrupted);
 
+private:
+  LockStack _lock_stack;
+
+public:
+  LockStack& lock_stack() { return _lock_stack; }
+
+  static ByteSize lock_stack_offset()      { return byte_offset_of(JavaThread, _lock_stack); }
+  static ByteSize lock_stack_top_offset()  { return lock_stack_offset() + LockStack::top_offset(); }
+  static ByteSize lock_stack_base_offset() { return lock_stack_offset() + LockStack::base_offset(); }
```

Comment:
Some commentary about why the offsets are all-defined relative to the base of the JavaThread would be nice.